iohomework/src/main/java/SalesExporter.java

```diff
@@ -0,0 +1,57 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * SalesExporter class is used to generate the output file and to write the statistics.
+ */
+
+class SalesExporter {
+
+    void writeStatsToFile(SalesStatistics stats) throws IOException {
+        File f = new File("stats.txt");
+        if (f.exists()) {
+            f.delete();
+        } else {
+            f.createNewFile();
+        }
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(f))) {
+            List<Product> products = stats.statsUnitsSold(stats.productSalesHashMap);
```

Comment:
Why are you passing encapsulated data from stats back to itself?!

Reply:
pai statsUnitsSold are nevoie de un HashMap sa-l sorteze dupa unitsSold (iar rezultatul il salvez intr-un list), iar productSalesHashMap nu e sortat. Acuma ca scriu probabil ar fi fost mai bine sa sortez direct productSalesHashMap nu? cu:
 Map<String,Product> sorted = stats.productSalesHashMap.entrySet().stream()
                            .sorted(comparingByValue()).collect(toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
am incercat asa numai ca nu-mi merge exporterul si imi da un NullPointerException si nu mi-am dat seama de unde